File: tests/fixtures/supabase/supabase-non-atomic-replace-pattern-broken/02-experiences-replace.ts
import { supabase } from '../client.js';

interface ExperienceRow {
  company: string;
  title: string;
}

export async function replaceExperiences(userId: string, rows: ExperienceRow[]) {
  await supabase.from('experiences').delete().eq('user_id', userId);
  for (const row of rows) {
    await supabase.from('experiences').insert({ user_id: userId, ...row });
  }
}

File: tests/fixtures/supabase/supabase-non-atomic-replace-pattern-broken/02_experiences_replace.py
from lib.client import supabase


def replace_experiences(user_id: str, rows: list[dict]):
    try:
        supabase.table("experiences").delete().eq("user_id", user_id).execute()
    except Exception:
        pass

    for row in rows:
        try:
            supabase.table("experiences").insert({"user_id": user_id, **row}).execute()
        except Exception:
            pass
